highstreets/features/processing_functions.py

```diff
@@ -0,0 +1,154 @@
+import pandas as pd
+from sklearn.linear_model import LinearRegression
+
+
+# create function to take averages of every x months
+# and see which hs fall in which proportion x months after that
+def create_labels(hs, label_month, comparison_start, comparison_end):
+    """Labels high streets at risk, depending on whether
+    spending is in which quartile of average of x months prior"""
+    hs_latest = hs.loc[(hs["month_year"] == label_month)]
+    hs_latest = hs_latest.groupby(["highstreet_name"]).mean()
+    # turn comp start and comp end variables into dates and find sequence btween
+    comparison_start = pd.to_datetime(comparison_start)
+    comparison_end = pd.to_datetime(comparison_end)
+    date_range = pd.date_range(comparison_start, comparison_end, freq="m").to_period(
+        "M"
+    )
+    hs_compare = hs.loc[(hs["month_year"]).isin(date_range)]
+    hs_compare = hs_compare.groupby(["highstreet_name"]).mean()
+
+    # merge dataframe
+    merged_df = pd.merge(hs_latest, hs_compare, on="highstreet_name")
+    # condtional labelling, if latest value not at least at average level, label at risk
+    merged_df["proportion"] = (
+        merged_df["txn_amt_wd_retail_x"] / merged_df["txn_amt_wd_retail_y"]
+    )
+    merged_df["labels"] = pd.qcut(
```

Comment:
Could this be simplified to pd.qcut(merged_df["proportion"], 4, labels=[0,1,2,3]) ?

Reply:
I'm also not completely clear how this achieves the functionality the function docstring describes. hs_latest will contain the means of all columns in the initial data frame averaged by highstreet for the label month, and hs_compare will contain means by highstreet during the comparison period for all columns. then the merged data frame takes a proportion of two particular columns and does a qcut on that proportion. Where does the latest value get compared to the comparison period explicitly?